### PR TITLE
Option to stop danger-xcodebuild from posting messages

### PR DESCRIPTION
**Pull request by ky1ejs**
``` ruby
xcodebuild.parse_warnings(post_message: true)
xcodebuild.parse_errors(post_message: true)
xcodebuild.parse_tests(post_message: true)
xcodebuild.perfect_build(post_message: true)
```

### CODE REVIEW

**Comment by valeriomazzeo** [comment]
@kylejm thank you for the pull request.

I was thinking, as they are all very similar and usually you either want the message or not in your configuration, maybe an instance variable would make more sense in this case?

Also, could you please add test coverage?

Thanks :)

**Comment by ky1ejs** [comment]
@valeriomazzeo I'm happy with either, which ever you prefer. The way I've implemented it offers more granular control, but the way you suggest likely makes more sense for users. As you say, they want them or they don't.

Which would you prefer?

**Comment by valeriomazzeo** [comment]
What is the specific use case you are encountering?

As we are not going to run this concurrently, in the worst case the user could change that variable in between calls to achieve a different result, so I would go for the variable (defaulting to `true`) instead of the parameter :+1:

**Comment by ky1ejs** [comment]
All done 😄 🍻

File: spec/xcodebuild_spec.rb [hunk 1]
```diff
@@ -137,6 +137,51 @@ module Danger
...
+          end
+
+          it "must not post messages" do
+              @xcodebuild.post_message = false
+              @xcodebuild.parse_warnings
+              @xcodebuild.parse_errors
+              @xcodebuild.parse_tests
+              @xcodebuild.perfect_build
+              expect(@dangerfile.status_report[:errors]).to be_empty
+              expect(@dangerfile.status_report[:errors]).to be_empty
```
Comment (valeriomazzeo) [inline spec/xcodebuild_spec.rb]: this seems to be repeated twice

File: README.md [hunk 2]
```diff
@@ -12,10 +12,17 @@ Exposes warnings, errors and test results. It requires a JSON generated using [x
...
+    xcodebuild.parse_errors # returns number of errors
+    xcodebuild.parse_tests # returns number of test failures
+    xcodebuild.perfect_build # returns a bool indicating if the build was perfect
+
+Messages are posted by default. You can stop them like this:
+
+    xcodebuild.parse_warnings(post_message: false)
+    xcodebuild.parse_errors(post_message: false)
+    xcodebuild.parse_tests(post_message: false)
+    xcodebuild.perfect_build(post_message: false)
```
Comment (valeriomazzeo) [inline README.md]: I think these changes can be reverted now

File: spec/xcodebuild_spec.rb [hunk 3]
```diff
@@ -137,6 +137,51 @@ module Danger
...
+              expect(@dangerfile.status_report[:warnings]).to be_empty
+              expect(@dangerfile.status_report[:messages]).to be_empty
+              expect(@dangerfile.status_report[:markdowns]).to be_empty
```
Comment (valeriomazzeo) [inline spec/xcodebuild_spec.rb:186]: Maybe we should add expectations for checking error and warning counts are correct?

**Review by valeriomazzeo — CHANGES_REQUESTED** [review]
Just some minor changes and it will be good to go.

Thank you for your contribution 👍🏻

File: spec/xcodebuild_spec.rb (hunk 1 above)
Comment (ky1ejs) [reply spec/xcodebuild_spec.rb]: Fixed 👍

File: spec/xcodebuild_spec.rb (hunk 3 above)
Comment (ky1ejs) [reply spec/xcodebuild_spec.rb:186]: Added 👍

File: README.md (hunk 2 above)
Comment (ky1ejs) [reply README.md]: Corrected 👍

**Review by valeriomazzeo — APPROVED** [review]

**Comment by valeriomazzeo** [comment]
looking good 👍

**Comment by ky1ejs** [comment]
@valeriomazzeo thanks for merging! Please could you release this now? 😊

**Comment by valeriomazzeo** [comment]
done 👍

**Pull request merged**